dartagnan/src/main/java/com/dat3m/dartagnan/program/event/core/AbstractEvent.java

```diff
@@ -258,8 +259,8 @@ public int compareTo(Event e) {
     }
 
     @Override
-    public Event getCopy() {
-        throw new UnsupportedOperationException("Copying is not allowed for " + getClass().getSimpleName());
+    public void transformExpressions(ExpressionVisitor<? extends Expression> exprTransformer) {
+        // WARNING: This should get overwritten by any event that reads Expressions
```

Comment:
Why not to add an exception then? Right now we will simply miss if an implementation is missing

Reply:
A lot of events actually have an empty implementation for that, for example `Fence`, `CodeAnnotation`,`Label`, and the SVCOMP atomic block events. I can add the exception, but then I need to provide an empty implementation for all of these events separately.
An alternative is to move this method into `RegReader`, assuming that all expression-dependent events are actually `RegReaders` (which I think is a fair assumption). I will move it.